beard/clustering/wrappers.py

```diff
@@ -168,31 +179,38 @@ def fit(self, X, y=None):
                                       criterion=self.criterion,
                                       depth=self.depth, R=self.R,
                                       monocrit=self.monocrit)
+                n_labels = len(np.unique(labels))
+
                 if ground_truth:
                     train = (y != -1)
 
-                    if not self.affinity_score:
-                        score = scoring(y[train], labels[train])
+                    if self.scoring_data == "row":
+                        score = scoring(Xr, y[train], labels[train])
+
+                    elif self.scoring_data == "affinity":
+                        score = scoring(Xa, y[train], labels[train])
+
                     else:
-                        score = scoring(Xs, y[train], labels[train])
+                        score = scoring(y[train], labels[train])
+
+                elif 1 < n_labels < n_samples:
```

Comment:
Why do we need this check actually? It seems you are making assumptions about the scoring function by doing that